### PR TITLE
Ensure that /var/run/reboot-needed is present on 102

### PR DESCRIPTION
**Pull request by mssola**
Sometimes zypper does not create this file when 102 is returned by `zypper
patch`. This commit forces the creation of this file for such cases.

Signed-off-by: Miquel Sabaté Solà <msabate@suse.com>

### CODE REVIEW

File: skuba_update/skuba_update.py [hunk 1]
```diff
@@ -71,6 +76,14 @@ def are_patches_available(code):
     return code == 100 or code == 101
 
 
+def is_reboot_needed(code):
+    """
+    Returns true of the given code is defined by zypper to mean that reboot is
+    needed.
+    """
+    return code == 102
```
Comment (davidcassany) [inline skuba_update/skuba_update.py]: I would take the chance to add as constants the return codes we are checking and make the code more explicit without _magic numbers_.

Probably we could use the same names as in zypper's man pages (e.g. ZYPPER_EXIT_INF_REBOOT_NEEDED)

File: skuba_update/skuba_update.py [hunk 2]
```diff
@@ -17,12 +17,17 @@
...
+from pathlib import Path
 
 # Since zypper 1.14.0, it will automatically create a /var/run/reboot-required
 # text file whenever one of the applied patches requires the system to be
 # rebooted. This will be used later on by kured.
 REQUIRED_ZYPPER_VERSION = (1, 14, 0)
 
+# The path to the reboot-needed file. This is the file that kured will be
+# looking at.
+REBOOT_NEEDED_PATH = '/var/run/reboot-needed'
```
Comment (ereslibre) [inline skuba_update/skuba_update.py]: Given that `kured` will only look for `/var/run/reboot-required` by default I think we can instead touch this file and ignore `/var/run/reboot-needed`.

File: skuba_update/skuba_update.py [hunk 3]
```diff
@@ -97,6 +110,13 @@ def run_zypper_command(command):
     process.communicate()
     if is_zypper_error(process.returncode):
         raise Exception('"{0}" failed'.format(' '.join(command)))
+
+    # In some cases zypper might return an exit code asking for a reboot, but
+    # at the same time it won't touch the REBOOT_NEEDED_PATH file. Let's make
+    # sure that this file is always created.
```
Comment (davidcassany) [inline skuba_update/skuba_update.py]: Probably I would express it slightly different, I'd say we trust the return code over the presence or not of the `/var/run/reboot-required` (which is different than REBOOT_NEEDED_PATH). But feel free to leave it as is if you prefer.

File: test/skuba_update_test.py [hunk 4]
```diff
@@ -148,3 +149,19 @@ def test_run_zypper_command_failure(mock_subprocess):
         exception = True
         assert '"zypper patch" failed' in str(e)
     assert exception
+
+
+@patch('subprocess.Popen')
+def test_run_zypper_command_creates_file_on_102(mock_subprocess):
+    mock_process = Mock()
+    mock_process.communicate.return_value = (None, None)
+    mock_process.returncode = 102
```
Comment (davidcassany) [inline test/skuba_update_test.py]: We could also import return code constants here and there would be no duplication and the code would be always consistent.

**Comment by mssola** [comment]
@ereslibre @davidcassany review again.

**Pull request merged**